crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -693,7 +693,7 @@ where
     }
```

Comment:
Note: above, there is an unnecessary `Sync` bound on the `Debug` implementation. I'm removing that impl entirely in #9588, so I didn't modify it to avoid conflicts.